node/src/main/scala/coop/rchain/node/grpc.scala

```diff
@@ -4,15 +4,20 @@ import io.grpc.{Server, ServerBuilder}
 import scala.concurrent.{ExecutionContext, Future}
 import cats._, cats.data._, cats.implicits._
 import coop.rchain.node.repl._
+import coop.rchain.rholang.interpreter.{RholangCLI, Runtime}
+import coop.rchain.rholang.interpreter.storage.StoragePrinter
+import monix.execution.Scheduler
 
-class GrpcServer(executionContext: ExecutionContext, port: Int) { self =>
+import java.io.StringReader
+
+class GrpcServer(executionContext: ExecutionContext, port: Int, runtime: Runtime) { self =>
 
   var server: Option[Server] = None
 
   def start(): Unit = {
```

Comment:
Just a thought -- what happens if you call `start()` twice?

Reply:
blows up bcs port is used

Reply:
How about using phantom types to prevent from that? Unless it's very unlikely to happen and the effort estimated is too big.